features/253-auth-sdk-callback-in-next/api.md

```diff
@@ -0,0 +1,393 @@
+}
+```
+
+New:
+```go=
+// Meta is meta data part of a user profile record
+type Meta struct {
+  ID         string                 `json:"_id"`
+  Type       string                 `json:"_type"`
+  RecordID   string                 `json:"_recordID"`
+  RecordType string                 `json:"_recordType"`
+  Access     map[string]interface{} `json:"_access"`
+  OwnerID    string                 `json:"_ownerID"`
+  CreatedAt  time.Time              `json:"_createdAt"`
+  CreatedBy  string                 `json:"_createdBy"`
+  UpdatedAt  time.Time              `json:"_updatedAt"`
+  UpdatedBy  string                 `json:"_updatedBy"`
+}
+
+// Data refers the profile info of a user,
+// like username, email, age, phone number
+type Data map[string]interface{}
+
+// UserProfile refers user profile data type
+type UserProfile struct {
+  Meta
+  Data
+}
+
+type Store interface {
+  CreateUserProfile(userID string, data Data) (UserProfile, error)
+  GetUserProfile(userID string) (UserProfile, error)
+}
+```
+
+And, use `NewAuthResponse` to generate auth response,
+
+Old:
+
+```go=
+func NewAuthResponse(authInfo authinfo.AuthInfo, user skydb.Record, accessToken string) AuthResponse
+```
+
+New:
+
+```go=
+func NewAuthResponse(authInfo authinfo.AuthInfo, userProfile userprofile.UserProfile, accessToken string) AuthResponse
+```
+
+# More usage example
+
+## login
+
+```javascript=
+skygear.auth.signupWithUsername(username, password).then((user) => {
+  console.log(user); // user is an user record
+  console.log(user["username"]); // username of the user
+}, (error) => {
+  ;
+});
+```
+
+## update user profile
+
+```javascript=
+var user = skygear.auth.currentUser;
+user["username"] = "new-username";
+skygear.auth.updateUser(user).then((user) => {
+  console.log(user); // updated user record
+  console.log('Username is changed to: ', user["username"]);
+  return skygear.auth.whoami();
+}, (error) => {
+  console.error(error);
+});
+```
+
+## update user profile via record gear (the error of permission denied is expected)
+
+```javascript=
+var user = skygear.auth.currentUser;
+user["username"] = "new-username";
+skygear.publicDB.save(user).then((user) => {
+  ;
+}, (error) => {
+  // permission denied
+  console.error(error);
+});
+```
+
+# access current user profile
+
+```javascript=
+console.log(skygear.auth.currentUser["username"]);
+console.log(skygear.auth.currentUser["email"]);
+console.log(skygear.auth.currentUser["gender"]);
+```
+
+# access current user auth info
+
+```javascript=
+console.log(skygear.auth.isCurrentUserVerified);
+console.log(skygear.auth.isCurrentUserDisabled);
+```
+
+## get current user roles
+
+```javascript=
+const users = [skygear.auth.currentUser];
+skygear.auth.fetchUserRole(skygear.auth.currentUser).then((roleMap) => {
+    console.log(roleMap); // { "<currentUsrId>": ["admin", "editor"] }
+}, (error) => {
+    console.error(error);
+});
+```
+
+## access aother user's profile
```

Comment:
Is the ACL for other user's profile same as Record's ACL?

Reply:
As I know, they should follow same policy.